### PR TITLE
style(weave): Facet panel styling

### PR DESCRIPTION
**Pull request by jwlee64**
adds axis labels
restyles select and column names
changes the way facet panels are sized

https://wandb.atlassian.net/browse/WB-14214

Screenshot 2023-07-27 at 9 39 46 AM

https://github.com/wandb/weave/assets/25037002/b5514331-9537-4dc6-9fb7-2eac93ccedde

### CODE REVIEW

File: weave/panels/panel_facet.py
Comment (tssweeney) [inline weave/panels/panel_facet.py:17]: please follow the convention of the other properties and make these nodes, not strings

File: weave-js/src/components/Panel2/PanelFacet/common.tsx
Comment (tssweeney) [inline weave-js/src/components/Panel2/PanelFacet/common.tsx:213]: the dash and lower case looks odd to me

**Review by tssweeney — CHANGES_REQUESTED** [review]
Main feedback here is that we have bailed out of Weave - please refactor and follow the weave-style conventions for node-based config fields.

File: weave-js/src/components/Panel2/PanelFacet/common.tsx
Comment (jwlee64) [reply weave-js/src/components/Panel2/PanelFacet/common.tsx]: synced with yangyi we going to go with "X-axis label" and just upper case X and Y

File: weave-js/src/components/Panel2/PanelFacet/Component.tsx
Comment (tssweeney) [inline weave-js/src/components/Panel2/PanelFacet/Component.tsx:431]: Must we require a label?

Comment (jwlee64) [reply weave-js/src/components/Panel2/PanelFacet/Component.tsx:431]: oops nope

File: weave/panels/panel_facet.py
Comment (tssweeney) [inline weave/panels/panel_facet.py:17]: Curious if you thought about putting these settings inside `FacetConfig`. It seems like the pattern for `dims` are to be fields in the resulting table. But labels apply to the whole table.

File: weave-js/src/components/Panel2/PanelFacet/Component.tsx
Comment (tssweeney) [inline weave-js/src/components/Panel2/PanelFacet/Component.tsx:387]: Was dropping the resize intentional?

Comment (jwlee64) [reply weave-js/src/components/Panel2/PanelFacet/Component.tsx:387]: yes resize is now intended to scale directly with the overall panel size

**Review by tssweeney — COMMENTED** [review]
Added some comments

File: weave/panels/panel_facet.py
Comment (jwlee64) [reply weave/panels/panel_facet.py:17]: I felt like grouping them with x and y fields directly seemed intuitive to me. They appear next to the x and y expressions in the config and grouping the x y fields with the label made sense

Comment (jwlee64) [reply weave/panels/panel_facet.py:17]: Not particularly opinionated about it tho

Comment (tssweeney) [reply weave/panels/panel_facet.py:17]: I can see your point here. Another perspective is that `FacetDimsConfig` is the config for the facet _dimensions_... so when computing the facet data, we construct a table where each column is defined by the dimensions defined in the FacetDimsConfig. However, in this pattern, we do that for 4 of the fields, but not the labels. The labels are calculated across the entire input, as opposed to row-level. I'll approve and leave this up to you. Consider that if we break this pattern, what is left to differentiate the FacetDimsConfig from the FacetConfig? One could argue at that point, everything should just go into the top level config (unless you have another property that serves to logically separate the DimsConfig

**Review by tssweeney — APPROVED** [review]
Approving with optional followup here: https://github.com/wandb/weave/pull/210/files#r1280957557

File: weave/panels/panel_facet.py
Comment (tssweeney) [reply weave/panels/panel_facet.py:17]: Once we commit to a config pattern, we must maintain support (or migrate) forever in the future - which i why i am being a bit careful here.

**Pull request merged**
